### DISCUSSION TITLE
Make reserved keywords escapeable

### DISCUSSION THREAD

**Discussion by dennisschroeder** [gleam-lang/gleam#1642] [Ideas & suggestions]
Right now, it is not possible to declare custom types where field names match a keyword.
Example:

```gleam 
    pub type Event {
        Event(type: String)
    }
```
The compiler will respond with a `syntax error` because `type` is a reserved keyword in gleam.
One solution could be to mark those keywords with single quotes `'type'`, since they have no special meaning in gleam, as far as I know.

What do you think about this?

**Comment by inoas** [comment]
If the call is no, I would love to have a definite list of all keywords not available for various purposes (like labels or fields) - would that be possible?

**Comment by Aloso** [comment]
Kotlin uses backticks:

```kt
fun `fun`() = "Very fun"
fun `this is also a function name`() = "Hello world"
```

Rust uses a `r#` prefix:

```rs
struct r#for {
    r#while: r#in,
}
```

Note that Kotlin allows special characters (like whitespace and punctuation) in identifiers, but Rust does not.
